File: Eulerjl/Euler3.jl
best = 0
num = 600851475143

function is_prime(x)
	if x%2 == 0
		return false
	end
	for i in 3:round(x^.5)+1
		if(x%i == 0)
			return false
		end
	end
	return true
end

for j in 3:round(num^.5)
	if(is_prime(j) && (num%j == 0))
		best = j
	end
end
println(best)

File: Eulercpp/Euler3.cpp
#include<iostream>
#include<math.h>
using namespace std;

bool isprime( long int n) {
    if(n < 2) {
        return false;
    }
	if(n == 2) {
		return true;
	}
	if(n % 2 == 0) {
		return false;
	}
	for(long int i = 3; i < sqrt(n) + 1; i += 2) {
		if(n % i == 0) {
			return false;
		}
	}
	return true;
}

int main(void) {
	long int num =  600851475143;
	long int best = 0;
	for(long int i = 1; i < num + 1; i++) {
		if(isprime(i) && num % i == 0) {
			best = i;
		}
	}
	cout << best << endl;
	return 0;
}
